js/app.tsx

```diff
@@ -0,0 +1,6 @@
+import "./phoenix.js";
+import { App } from "./components/app.js";
+import { createRoot } from "react-dom/client";
+
+const root = createRoot(document.getElementById("app") as Element);
```

Comment:
**suggestion:** I slightly prefer writing this as:

```suggestion
const root = createRoot(document.getElementById("app")!);
```

The `!` specifically asserts that the value is non-null, so it makes it a bit clearer that it's narrowing down `Element | null` as opposed to some other `Element | Foo` type that you might also use an `as` assertion on.

Reply:
Ah, gotcha! Thanks

Reply:
Oh, hah, by default the ESLint rules forbid non-null assertions but not `as` assertions, despite the latter being IMO more dangerous. I think adding an ESLint override would be appropriate here.